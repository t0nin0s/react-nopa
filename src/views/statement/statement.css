.desktop-only {
  display: none;
}

.button {
  width: 95%;
}

@media all and (min-width: 360px) {
  .container {
    width: 95%;
    max-width: 40rem;
    margin: 0 auto;
  }

  .desktop-only {
    display: block;
  }

  .button {
    width: 45%;
  }
}
