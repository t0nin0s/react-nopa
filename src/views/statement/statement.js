import React from 'react'

import Layout from '../../components/layout/layout'
import AccountDetail from '../../components/accountDetail/accountDetail'
import TransactionsTable from '../../components/transactionsTable/transactionsTable'

import styles from './statement.css'

export const OUT = 'OUT'
export const IN = 'IN'

const transactions = [
  {date: 'Today', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: 'Today', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: 'Today', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '22 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '22 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '21 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '21 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '20 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
]

const Statement = () => (
  <Layout>
    <div className='statement'>
      <div className='content'>
        <section className='section section--primary'>
          <div className={[styles.container, 'centered'].join(' ')}>
            <h1 className={[styles.desktopOnly, 'section__h1'].join(' ')}>
              Which bank does this account belong to?
            </h1>
            <p className={[styles.desktopOnly, 'section__p'].join(' ')}>
              {`Track of all your payments by connecting as
              many bank accounts as you'd like to your
              NOPA account and get updates your balance
              instantly. Plus it's free.`}
            </p>
            <AccountDetail />
            <TransactionsTable transactions={transactions}/>
            <a href="#" className={[styles.button, 'button'].join(' ')}>Show More</a>
          </div>
        </section>
      </div>
    </div>
  </Layout>
)

export default Statement
