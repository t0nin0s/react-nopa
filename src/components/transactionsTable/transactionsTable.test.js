import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import TransactionsTable from './transactionsTable'
import { IN, OUT } from '../../views/statement/statement'

const transactions = [
  {date: 'Today', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: 'Today', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: 'Today', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '22 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '22 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '21 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '21 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
  {date: '20 November 2016', description: 'Tesco', type: OUT, amount: '35.98'},
]

test('TransactionsTable component renders as expected', () => {
  const component = shallow(<TransactionsTable  transactions={transactions} />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
