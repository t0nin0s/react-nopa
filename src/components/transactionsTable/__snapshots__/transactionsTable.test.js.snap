// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsTable component renders as expected 1`] = `
<div
  className="wrapper"
>
  <span>
    Your transactions for the last 30 days
  </span>
  <div
    className="transactionTable"
  >
    <TransactionCategoryRow
      category="Today"
    />
    <TransactionRow
      amount="35.98"
      description="Tesco"
      type="OUT"
    />
    <TransactionRow
      amount="35.98"
      description="Tesco"
      type="OUT"
    />
    <TransactionRow
      amount="35.98"
      description="Tesco"
      type="OUT"
    />
    <TransactionCategoryRow
      category="22 November 2016"
    />
    <TransactionRow
      amount="35.98"
      description="Tesco"
      type="OUT"
    />
    <TransactionRow
      amount="35.98"
      description="Tesco"
      type="OUT"
    />
    <TransactionCategoryRow
      category="21 November 2016"
    />
    <TransactionRow
      amount="35.98"
      description="Tesco"
      type="OUT"
    />
    <TransactionRow
      amount="35.98"
      description="Tesco"
      type="OUT"
    />
    <TransactionCategoryRow
      category="20 November 2016"
    />
    <TransactionRow
      amount="35.98"
      description="Tesco"
      type="OUT"
    />
  </div>
</div>
`;
